## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-erasure.sh | :x: [more...](#ebb4cafc-a43d-4a5f-8c96-e633f36ca578) |
| mint-pools.sh | :x: [more...](#f5c53ec6-2085-407b-9d85-4844822571cd) |
<a name="ebb4cafc-a43d-4a5f-8c96-e633f36ca578"></a>
### 16859-6b4666c/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:30537
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 39ba5c804840:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... done in 1 minutes and 14 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 15 seconds
(8/14) Running minio-go tests ... FAILED in 1 minutes and 22 seconds
{
  "args": {
    "bucketName": "minio-go-test-y4bgop6vwtygvlhd",
    "fileName": "/mint/data/datafile-129-MB",
    "location": "us-east-1",
    "objectName": "minio-go-test-y4bgop6vwtygvlhdFPutObject-standard",
    "opts": {
      "UserMetadata": null,
      "UserTags": null,
      "Progress": null,
      "ContentType": "application/octet-stream",
      "ContentEncoding": "",
      "ContentDisposition": "",
      "ContentLanguage": "",
      "CacheControl": "",
      "Mode": "",
      "RetainUntilDate": "0001-01-01T00:00:00Z",
      "ServerSideEncryption": null,
      "NumThreads": 0,
      "StorageClass": "",
      "WebsiteRedirectLocation": "",
      "PartSize": 0,
      "LegalHold": "",
      "SendContentMd5": false,
      "DisableContentSha256": false,
      "DisableMultipart": false,
      "Internal": {
        "SourceVersionID": "",
        "SourceETag": "",
        "ReplicationStatus": "",
        "SourceMTime": "0001-01-01T00:00:00Z",
        "ReplicationRequest": false,
        "RetentionTimestamp": "0001-01-01T00:00:00Z",
        "TaggingTimestamp": "0001-01-01T00:00:00Z",
        "LegalholdTimestamp": "0001-01-01T00:00:00Z"
      }
    }
  },
  "duration": 60534,
  "error": "The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.",
  "function": "FPutObject(bucketName, objectName, fileName, opts)",
  "message": "FPutObject failed",
  "name": "minio-go: testFPutObject",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... done in 17 seconds
(9/14) Running minio-js tests ... done in 46 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 26 seconds
(11/14) Running s3cmd tests ... done in 16 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 8 seconds

Executed 13 out of 14 tests successfully.
```
<a name="f5c53ec6-2085-407b-9d85-4844822571cd"></a>
### 16859-6b4666c/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.2:31900
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp f6f1461e7da5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 9 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 1 minutes and 17 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 18 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 50 seconds
(9/14) Running minio-java tests ... done in 37 seconds
(10/14) Running minio-js tests ... done in 52 seconds
(11/14) Running minio-py tests ... FAILED in 10 minutes and 14 seconds
{
  "name": "minio-py:test_list_object_versions",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-9c0a8dc8-75e4-44e9-9cf3-849e2f0afbef",
    "object_name": "eeb223aa-687e-413b-be6e-0a29f40498bc",
    "recursive": true
  },
  "message": "S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-9c0a8dc8-75e4-44e9-9cf3-849e2f0afbef, request_id: 174E419502CA8445, host_id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, bucket_name: minio-py-test-9c0a8dc8-75e4-44e9-9cf3-849e2f0afbef",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1253, in test_list_object_versions\n    _test_list_objects(log_entry, version_check=True)\n  File \"/mint/run/core/minio-py/tests.py\", line 1064, in _test_list_objects\n    _CLIENT.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-9c0a8dc8-75e4-44e9-9cf3-849e2f0afbef, request_id: 174E419502CA8445, host_id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, bucket_name: minio-py-test-9c0a8dc8-75e4-44e9-9cf3-849e2f0afbef\n",
  "duration": 120141
}
(11/14) Running s3cmd tests ... FAILED in 4 seconds
{
  "name": "s3cmd",
  "duration": "3376",
  "function": "setup",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 3 sec...\nERROR: S3 error: 409 (BucketAlreadyOwnedByYou): Your previous request to create the named bucket succeeded and you already own it."
}
(11/14) Running s3select tests ... FAILED in 8 minutes and 11 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-69d1ab46-0b8a-4ac4-8727-dde42f26d045"
  },
  "duration": 490973,
  "message": "Test test_1 unexpectedly failed with: HTTPConnectionPool(host='15.15.15.2', port=31900): Max retries exceeded with url: /s3select-test-69d1ab46-0b8a-4ac4-8727-dde42f26d045/c41bca38-bd07-4f3f-8e87-568b23ab4bd2 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/s3select/csv.py\", line 37, in test_sql_api\n    client.put_object(bucket_name, object_name,\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1766, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1720, in put_object\n    return self._put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1578, in _put_object\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 1 more time]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.2', port=31900): Max retries exceeded with url: /s3select-test-69d1ab46-0b8a-4ac4-8727-dde42f26d045/c41bca38-bd07-4f3f-8e87-568b23ab4bd2 (Caused by ResponseError('too many 503 error responses'))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 117, in test_csv_input_custom_quote_char\n    test_sql_api(f'test_{idx}', client, bucket_name,\n  File \"/mint/run/core/s3select/csv.py\", line 47, in test_sql_api\n    raise ValueError(\nValueError: Test test_1 unexpectedly failed with: HTTPConnectionPool(host='15.15.15.2', port=31900): Max retries exceeded with url: /s3select-test-69d1ab46-0b8a-4ac4-8727-dde42f26d045/c41bca38-bd07-4f3f-8e87-568b23ab4bd2 (Caused by ResponseError('too many 503 error responses'))\n",
  "status": "FAIL"
}
(11/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-6oyk0f6fhidzf4"
  },
  "duration": 175,
  "error": "NoSuchBucket: The specified bucket does not exist\n\tstatus code: 404, request id: 174E42089895DC20, host id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
  "function": "/mint/build/versioning/bucket.go:76: main.testMakeBucket",
  "message": "Get Versioning failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 10 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/admin-handlers-users.go
+++ b/cmd/admin-handlers-users.go
@@ -1956,7 +1956,7 @@ func (a adminAPIHandlers) DetachPolicyBuiltin(w http.ResponseWriter, r *http.Req
 			UserOrGroup: userOrGroup,
 			UserType:    int(userType),
 			IsGroup:     isGroup,
-			Policy:      strings.Join(policiesToDetach, ","),
+			Policy:      newPolicies,
 		},
 		UpdatedAt: updatedAt,
 	}))

--- a/cmd/peer-s3-client.go
+++ b/cmd/peer-s3-client.go
@@ -167,7 +167,7 @@ func (sys *S3PeerSys) GetBucketInfo(ctx context.Context, bucket string, opts Buc
 
 	quorum := (len(sys.allPeerClients) / 2)
 	if err = reduceReadQuorumErrs(ctx, errs, bucketOpIgnoredErrs, quorum); err != nil {
-		return BucketInfo{}, err
+		return BucketInfo{}, toObjectErr(err, bucket)
 	}
 
 	for i, err := range errs {
@@ -266,12 +266,9 @@ func (sys *S3PeerSys) DeleteBucket(ctx context.Context, bucket string, opts Dele
 	errs := g.Wait()
 	errs = append(errs, deleteBucketLocal(ctx, bucket, opts))
 
-	for _, err := range errs {
-		if err != nil {
-			return err
-		}
-	}
-	return nil
+	quorum := (len(sys.allPeerClients) / 2) + 1
+	err := reduceWriteQuorumErrs(ctx, errs, bucketOpIgnoredErrs, quorum)
+	return toObjectErr(err, bucket)
 }
 
 // DeleteBucket deletes bucket on a peer

--- a/docs/bucket/replication/setup_replication.sh
+++ b/docs/bucket/replication/setup_replication.sh
@@ -38,11 +38,11 @@ cat > repladmin-policy-source.json <<EOF
     ]
    }
 EOF
-mc admin policy add source repladmin-policy ./repladmin-policy-source.json
+mc admin policy create source repladmin-policy ./repladmin-policy-source.json
 cat ./repladmin-policy-source.json
 
 #assign this replication policy to repladmin
-mc admin policy set source repladmin-policy user=repladmin
+mc admin policy attach source repladmin-policy --user=repladmin
 
 ### on dest alias
 # Create a replication user : repluser on dest alias
@@ -90,11 +90,11 @@ cat > replpolicy.json <<EOF
  ]
 }
 EOF
-mc admin policy add dest replpolicy ./replpolicy.json
+mc admin policy create dest replpolicy ./replpolicy.json
 cat ./replpolicy.json
 
 # assign this replication policy to repluser
-mc admin policy set dest replpolicy user=repluser
+mc admin policy attach dest replpolicy --user=repluser
 
 # configure replication config to remote bucket at http://localhost:9000
 mc replicate add source/bucket --priority 1 --remote-bucket http://repluser:repluser123@localhost:9000/bucket \

--- a/docs/distributed/decom-compressed-sse-s3.sh
+++ b/docs/distributed/decom-compressed-sse-s3.sh
@@ -29,11 +29,11 @@ sleep 2
 ./mc admin user add myminio/ minio123 minio123
 ./mc admin user add myminio/ minio12345 minio12345
 
-./mc admin policy add myminio/ rw ./docs/distributed/rw.json
-./mc admin policy add myminio/ lake ./docs/distributed/rw.json
+./mc admin policy create myminio/ rw ./docs/distributed/rw.json
+./mc admin policy create myminio/ lake ./docs/distributed/rw.json
 
-./mc admin policy set myminio/ rw user=minio123
-./mc admin policy set myminio/ lake,rw user=minio12345
+./mc admin policy attach myminio/ rw --user=minio123
+./mc admin policy attach myminio/ lake,rw --user=minio12345
 
 ./mc mb -l myminio/versioned
 

--- a/docs/distributed/decom-encrypted-sse-s3.sh
+++ b/docs/distributed/decom-encrypted-sse-s3.sh
@@ -24,11 +24,11 @@ sleep 2
 ./mc admin user add myminio/ minio123 minio123
 ./mc admin user add myminio/ minio12345 minio12345
 
-./mc admin policy add myminio/ rw ./docs/distributed/rw.json
-./mc admin policy add myminio/ lake ./docs/distributed/rw.json
+./mc admin policy create myminio/ rw ./docs/distributed/rw.json
+./mc admin policy create myminio/ lake ./docs/distributed/rw.json
 
-./mc admin policy set myminio/ rw user=minio123
-./mc admin policy set myminio/ lake,rw user=minio12345
+./mc admin policy attach myminio/ rw --user=minio123
+./mc admin policy attach myminio/ lake,rw --user=minio12345
 
 ./mc mb -l myminio/versioned
 

--- a/docs/distributed/decom-encrypted.sh
+++ b/docs/distributed/decom-encrypted.sh
@@ -26,11 +26,11 @@ export MC_HOST_myminio="http://minioadmin:minioadmin@localhost:9000/"
 ./mc admin user add myminio/ minio123 minio123
 ./mc admin user add myminio/ minio12345 minio12345
 
-./mc admin policy add myminio/ rw ./docs/distributed/rw.json
-./mc admin policy add myminio/ lake ./docs/distributed/rw.json
+./mc admin policy create myminio/ rw ./docs/distributed/rw.json
+./mc admin policy create myminio/ lake ./docs/distributed/rw.json
 
-./mc admin policy set myminio/ rw user=minio123
-./mc admin policy set myminio/ lake,rw user=minio12345
+./mc admin policy attach myminio/ rw --user=minio123
+./mc admin policy attach myminio/ lake,rw --user=minio12345
 
 ./mc mb -l myminio/versioned
 

--- a/docs/distributed/decom.sh
+++ b/docs/distributed/decom.sh
@@ -25,11 +25,11 @@ export MC_HOST_myminio="http://minioadmin:minioadmin@localhost:9000/"
 ./mc admin user add myminio/ minio123 minio123
 ./mc admin user add myminio/ minio12345 minio12345
 
-./mc admin policy add myminio/ rw ./docs/distributed/rw.json
-./mc admin policy add myminio/ lake ./docs/distributed/rw.json
+./mc admin policy create myminio/ rw ./docs/distributed/rw.json
+./mc admin policy create myminio/ lake ./docs/distributed/rw.json
 
-./mc admin policy set myminio/ rw user=minio123
-./mc admin policy set myminio/ lake,rw user=minio12345
+./mc admin policy attach myminio/ rw --user=minio123
+./mc admin policy attach myminio/ lake,rw --user=minio12345
 
 ./mc mb -l myminio/versioned
 

--- a/docs/multi-user/README.md
+++ b/docs/multi-user/README.md
@@ -41,7 +41,7 @@ EOF
 Create new canned policy by name `getonly` using `getonly.json` policy file.
 
 ```
-mc admin policy add myminio getonly getonly.json
+mc admin policy create myminio getonly getonly.json
 ```
 
 Create a new user `newuser` on MinIO use `mc admin user`.
@@ -53,7 +53,7 @@ mc admin user add myminio newuser newuser123
 Once the user is successfully created you can now apply the `getonly` policy for this user.
 
 ```
-mc admin policy set myminio getonly user=newuser
+mc admin policy attach myminio getonly --user=newuser
 ```
 
 ### 3. Create a new group
@@ -65,7 +65,7 @@ mc admin group add myminio newgroup newuser
 Once the group is successfully created you can now apply the `getonly` policy for this group.
 
 ```
-mc admin policy set myminio getonly group=newgroup
+mc admin policy attach myminio getonly --group=newgroup
 ```
 
 ### 4. Disable user
@@ -107,13 +107,13 @@ mc admin group remove myminio newgroup
 Change the policy for user `newuser` to `putonly` canned policy.
 
 ```
-mc admin policy set myminio putonly user=newuser
+mc admin policy attach myminio putonly --user=newuser
 ```
 
 Change the policy for group `newgroup` to `putonly` canned policy.
 
 ```
-mc admin policy set myminio putonly group=newgroup
+mc admin policy attach myminio putonly --group=newgroup
 ```
 
 ### 7. List all users or groups

--- a/docs/multi-user/admin/README.md
+++ b/docs/multi-user/admin/README.md
@@ -50,7 +50,7 @@ EOF
 Create new canned policy by name `userManager` using `userManager.json` policy file.
 
 ```
-mc admin policy add myminio userManager adminManageUser.json
+mc admin policy attach myminio userManager adminManageUser.json
 ```
 
 Create a new admin user `admin1` on MinIO use `mc admin user`.
@@ -62,7 +62,7 @@ mc admin user add myminio admin1 admin123
 Once the user is successfully created you can now apply the `userManage` policy for this user.
 
 ```
-mc admin policy set myminio userManager user=admin1
+mc admin policy attach myminio userManager --user=admin1
 ```
 
 This admin user will then be allowed to perform create/delete user operations via `mc admin user`
@@ -73,8 +73,8 @@ This admin user will then be allowed to perform create/delete user operations vi
 mc alias set myminio-admin1 http://localhost:9000 admin1 admin123 --api s3v4
 
 mc admin user add myminio-admin1 user1 user123
-mc admin policy add myminio-admin1 user1policy ~/user1policy.json
-mc admin policy set myminio-admin1 user1policy user=user1
+mc admin policy attach myminio-admin1 user1policy ~/user1policy.json
+mc admin policy attach myminio-admin1 user1policy --user=user1
 ```
 
 ### 4. List of permissions defined for admin operations

--- a/docs/site-replication/run-multi-site-ldap.sh
+++ b/docs/site-replication/run-multi-site-ldap.sh
@@ -64,12 +64,12 @@ export MC_HOST_minio3=http://minio:minio123@localhost:9003
 
 ./mc admin replicate add minio1 minio2 minio3
 
-./mc admin policy set minio1 consoleAdmin user="uid=dillon,ou=people,ou=swengg,dc=min,dc=io"
+./mc admin policy attach minio1 consoleAdmin --user="uid=dillon,ou=people,ou=swengg,dc=min,dc=io"
 sleep 5
 
 ./mc admin user info minio2 "uid=dillon,ou=people,ou=swengg,dc=min,dc=io"
 ./mc admin user info minio3 "uid=dillon,ou=people,ou=swengg,dc=min,dc=io"
-./mc admin policy add minio1 rw ./docs/site-replication/rw.json
+./mc admin policy create minio1 rw ./docs/site-replication/rw.json
 
 sleep 5
 ./mc admin policy info minio2 rw >/dev/null 2>&1

--- a/docs/site-replication/run-multi-site-minio-idp.sh
+++ b/docs/site-replication/run-multi-site-minio-idp.sh
@@ -61,14 +61,14 @@ export MC_HOST_minio3=http://minio:minio123@localhost:9003
 ## add foobar-g group with foobar
 ./mc admin group add minio2 foobar-g foobar
 
-./mc admin policy set minio1 consoleAdmin user=foobar
+./mc admin policy attach minio1 consoleAdmin --user=foobar
 sleep 5
 
 ./mc admin user info minio2 foobar
 
 ./mc admin group info minio1 foobar-g
 
-./mc admin policy add minio1 rw ./docs/site-replication/rw.json
+./mc admin policy create minio1 rw ./docs/site-replication/rw.json
 
 sleep 5
 ./mc admin policy info minio2 rw >/dev/null 2>&1
@@ -299,7 +299,7 @@ if [ $? -ne 0 ]; then
     echo "adding user failed, exiting.."
     exit_1;
 fi
-./mc admin policy set minio1 consoleAdmin user=foobarx
+./mc admin policy attach minio1 consoleAdmin --user=foobarx
 if [ $? -ne 0 ]; then
     echo "adding policy mapping failed, exiting.."
     exit_1;
@@ -307,7 +307,7 @@ fi
 sleep 10
 
 # unset policy for foobarx in minio2
-./mc admin policy unset minio2 consoleAdmin user=foobarx
+./mc admin policy detach minio2 consoleAdmin --user=foobarx
 if [ $? -ne 0 ]; then
     echo "unset policy mapping failed, exiting.."
     exit_1;
@@ -318,10 +318,10 @@ fi
 
 sleep 10
 
-# Test whether policy unset replicated to minio1
+# Test whether policy detach replicated to minio1
 policy=$(./mc admin user info minio1 foobarx --json | jq -r .policyName)
 if [ "${policy}" != "null" ]; then
-    echo "expected policy unset to have replicated, exiting..."
+    echo "expected policy detach to have replicated, exiting..."
     exit_1;
 fi
 

--- a/docs/site-replication/run-multi-site-oidc.sh
+++ b/docs/site-replication/run-multi-site-oidc.sh
@@ -65,7 +65,7 @@ export MC_HOST_minio3=http://minio:minio123@localhost:9003
 
 ./mc admin replicate add minio1 minio2 minio3
 
-./mc admin policy add minio1 projecta ./docs/site-replication/rw.json
+./mc admin policy create minio1 projecta ./docs/site-replication/rw.json
 sleep 5
 
 ./mc admin policy info minio2 projecta >/dev/null 2>&1
@@ -94,7 +94,7 @@ if [ $? -eq 0 ]; then
     exit_1;
 fi
 
-./mc admin policy add minio1 projecta ./docs/site-replication/rw.json
+./mc admin policy create minio1 projecta ./docs/site-replication/rw.json
 sleep 5
 
 # Generate STS credential with STS call to minio1

--- a/docs/sts/dex.md
+++ b/docs/sts/dex.md
@@ -39,7 +39,7 @@ time="2020-07-12T20:45:50Z" level=info msg="listening (http) on 0.0.0.0:5556"
 ```
 
 ```
-~ mc admin policy add admin allaccess.json
+~ mc admin policy create admin allaccess.json
 ```
 
 Contents of `allaccess.json`
@@ -95,7 +95,7 @@ Now you have successfully configured Dex IdP with MinIO.
 export MINIO_IDENTITY_OPENID_CLAIM_NAME=groups
 ```
 
-and add relevant policies on MinIO using `mc admin policy add myminio/ <group_name> group-access.json`
+and add relevant policies on MinIO using `mc admin policy create myminio/ <group_name> group-access.json`
 
 ## Explore Further
 

--- a/docs/sts/ldap.md
+++ b/docs/sts/ldap.md
@@ -153,7 +153,7 @@ In the configuration variables, `%s` is substituted with the _username_ from the
 Access policies may be associated by their name with a group or user directly. Access policies are first defined on the MinIO server using IAM policy JSON syntax. To define a new policy, you can use the [AWS policy generator](https://awspolicygen.s3.amazonaws.com/policygen.html). Copy the policy into a text file `mypolicy.json` and issue the command like so:
 
 ```sh
-mc admin policy add myminio mypolicy mypolicy.json
+mc admin policy create myminio mypolicy mypolicy.json
 ```
 
 To associate the policy with an LDAP user or group, use the full DN of the user or group:
@@ -163,7 +163,7 @@ mc admin idp ldap policy attach myminio mypolicy --user='uid=james,cn=accounts,d
 ```
 
 ```sh
-mc admin idp ldap policy attach myminio mypolicy --group='cn=projectx,ou=groups,ou=hwengg,dc=min,dc=io'
+mc admin idp ldap policy attach myminio mypolicy ----group='cn=projectx,ou=groups,ou=hwengg,dc=min,dc=io'
 ```
 
 To remove a policy association, use the similar `detach` command:
@@ -173,7 +173,7 @@ mc admin idp ldap policy detach myminio mypolicy --user='uid=james,cn=accounts,d
 ```
 
 ```sh
-mc admin idp ldap policy detach myminio mypolicy --group='cn=projectx,ou=groups,ou=hwengg,dc=min,dc=io'
+mc admin idp ldap policy detach myminio mypolicy ----group='cn=projectx,ou=groups,ou=hwengg,dc=min,dc=io'
 ```
 
 
@@ -184,12 +184,12 @@ Note that the commands above attempt to validate if the given entity (user or gr
 Please **do not use** these as they may be removed or their behavior may change.
 
 ```sh
-mc admin policy set myminio mypolicy user='uid=james,cn=accounts,dc=myldapserver,dc=com'
+mc admin policy attach myminio mypolicy --user='uid=james,cn=accounts,dc=myldapserver,dc=com'
 ```
 
 
 ```sh
-mc admin policy set myminio mypolicy group='cn=projectx,ou=groups,ou=hwengg,dc=min,dc=io'
+mc admin policy attach myminio mypolicy --group='cn=projectx,ou=groups,ou=hwengg,dc=min,dc=io'
 ```
 
 </details>

--- a/helm/minio/templates/_helper_create_policy.txt
+++ b/helm/minio/templates/_helper_create_policy.txt
@@ -55,7 +55,7 @@ createPolicy () {
   else
     echo "Policy '$NAME' already exists."
   fi
-  ${MC} admin policy add myminio $NAME /config/$FILENAME.json
+  ${MC} admin policy create myminio $NAME /config/$FILENAME.json
 
 }
 

--- a/helm/minio/templates/_helper_create_user.txt
+++ b/helm/minio/templates/_helper_create_user.txt
@@ -73,7 +73,7 @@ createUser() {
   # set policy for user
   if [ ! -z $POLICY -a $POLICY != " " ] ; then
       echo "Adding policy '$POLICY' for '$USER'"
-      ${MC} admin policy set myminio $POLICY user=$USER
+      ${MC} admin policy attach myminio $POLICY --user=$USER
   else
       echo "User '$USER' has no policy attached."
   fi

--- a/helm/minio/values.yaml
+++ b/helm/minio/values.yaml
@@ -433,7 +433,7 @@ makeBucketJob:
 ## List of command to run after minio install
 ## NOTE: the mc command TARGET is always "myminio"
 customCommands:
-  # - command: "admin policy set myminio consoleAdmin group='cn=ops,cn=groups,dc=example,dc=com'"
+  # - command: "admin policy attach myminio consoleAdmin --group='cn=ops,cn=groups,dc=example,dc=com'"
 
 ## Additional Annotations for the Kubernetes Job customCommandJob
 customCommandJob:

